### DISCUSSION TITLE
Hi, i have problem about Quick login

### DISCUSSION THREAD

**Discussion by heyChenfu** [lupidan/apple-signin-unity#127] [General]
My project have multiple login way, I see the demo use QuickLogin when there is no apple user id storage, so QuickLogin is triggered when the player enters the login panel, is this correct? But there are multi login way, we don't know which one the player will use. If not, when should the QuickLogin trigger, or maybe no need to use that?
I know README have specific QuickLogin, but i don't quite understand yet, thanks

**Comment by lupidan** [comment]
Quick login is used to request possible existing credentials with Apple using your app. If the user has used Sign in with Apple in the past, and has not revoked those credentials, a login screen will be triggered, and you will get the Identity token and the Apple User ID.
If not, Quick login will just fail.

The example calls it on App Launch to mimic the official Apple [example](https://developer.apple.com/documentation/authenticationservices/implementing_user_authentication_with_sign_in_with_apple?language=objc). It´s up to you and your app if you want to call this method, and when.
